src/main/java/core/basesyntax/web/filter/AuthenticationFilter.java

```diff
@@ -0,0 +1,51 @@
+package core.basesyntax.web.filter;
+
+import core.basesyntax.lib.Injector;
+import core.basesyntax.service.DriverService;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AuthenticationFilter implements Filter {
+    private static final String DRIVER_ID = "driverId";
+    private static final Injector INJECTOR = Injector.getInstance("core.basesyntax");
+    private final DriverService driverService = (DriverService) INJECTOR
+            .getInstance(DriverService.class);
+    private final Set<String> urls = new HashSet<>();
```

Comment:
```suggestion
    private final Set<String> allowedUrls = new HashSet<>();
```

Reply:
Fixed. As for points in the chat - I don`t think that it`s a good idea to throw one exception in my case. I have different messages about wrong actions for the user.